core/src/muxing/boxed.rs

```diff
@@ -65,14 +59,16 @@ where
             .map_err(into_io_error)
     }
 
-    fn poll_address_change(
+    #[inline]
```

Comment:
Why inline this method but not the other ones, and why only inline it in this muxer? Just asking out of curiosity because I am not really familiar with `#[inline]`.

Reply:
Not on purpose to be honest. I thought I left everything as it was before.

I am not too familiar with inlining either but the rough advice I got was that the compiler tends to be smarter on when it is needed :)

Reply:
I think the many `#[inline]` attributes are from past premature optimizations. See https://github.com/libp2p/rust-libp2p/issues/897.

> the rough advice I got was that the compiler tends to be smarter on when it is needed :)

Yes. Unless proven through a benchmark, let the compiler make the decision and thus don't use `#[inline]`.

Reply:
> Not on purpose to be honest. I thought I left everything as it was before.

The method was marked as `#[inline]` before this patch as well. I am fine with either removing it here or in a pull request in the future.